### PR TITLE
EMBCESSMOD-5013: backend changes for needs assesment changes (#58)

### PR DESCRIPTION
**Pull request by ytqsl**

### CODE REVIEW

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Contract.cs [hunk 1]
```diff
@@ -170,6 +166,16 @@ public enum NeedsAssessmentType
         Assessed
     }
 
+    public enum IdentifiedNeed
+    {
+        ShelterReferral,
```
Comment (williamrosswhite) [inline ess/src/API/EMBC.ESS/Resources/Evacuations/Contract.cs:171]: I think these are supposed to be two options on one field, not two fields

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs [hunk 2]
```diff
@@ -67,11 +68,11 @@ public Mappings()
...
-                  .ForMember(d => d.era_canevacueeprovideincidentals, opts => opts.MapFrom(s => Lookup(s.CanProvideIncidentals)))
-                  .ForMember(d => d.era_canevacueeprovidelodging, opts => opts.MapFrom(s => Lookup(s.CanProvideLodging)))
-                  .ForMember(d => d.era_canevacueeprovidetransportation, opts => opts.MapFrom(s => Lookup(s.CanProvideTransportation)))
+                  .ForMember(d => d.era_canevacueeprovidefood, opts => opts.MapFrom(s => s.Needs.Contains(IdentifiedNeed.Food) ? (int)NeedTrueFalse.False : (int)NeedTrueFalse.True))
+                  .ForMember(d => d.era_canevacueeprovideclothing, opts => opts.MapFrom(s => s.Needs.Contains(IdentifiedNeed.Clothing) ? (int)NeedTrueFalse.False : (int)NeedTrueFalse.True))
+                  .ForMember(d => d.era_canevacueeprovideincidentals, opts => opts.MapFrom(s => s.Needs.Contains(IdentifiedNeed.Clothing) ? (int)NeedTrueFalse.False : (int)NeedTrueFalse.True))
+                  .ForMember(d => d.era_canevacueeprovidelodging, opts => opts.MapFrom(s => s.Needs.Contains(IdentifiedNeed.ShelterReferral) ? (int)NeedTrueFalse.False : (int)NeedTrueFalse.True))
```
Comment (williamrosswhite) [inline ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs:74]: I think in the backend /database there is no longer a canevacueeprovidelogding, and instead a new shelterOptions string field

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs [hunk 3]
```diff
@@ -96,15 +97,21 @@ public Mappings()
...
                 .ForMember(d => d.Pets, opts => opts.Ignore())
                 .ForMember(d => d.Notes, opts => opts.Ignore())
+                .ForMember(d => d.Needs, opts => opts.Ignore())
+                .AfterMap((s, d) =>
+                {
+                    var needs = new List<IdentifiedNeed>();
+                    if (s.era_canevacueeprovideclothing.GetValueOrDefault(0) == (int)NeedTrueFalse.False) needs.Add(IdentifiedNeed.Clothing);
+                    if (s.era_canevacueeprovidefood.GetValueOrDefault(0) == (int)NeedTrueFalse.False) needs.Add(IdentifiedNeed.Food);
+                    if (s.era_canevacueeprovideincidentals.GetValueOrDefault(0) == (int)NeedTrueFalse.False) needs.Add(IdentifiedNeed.Incidentals);
+                    if (s.era_canevacueeprovidelodging.GetValueOrDefault(0) == (int)NeedTrueFalse.False) needs.Add(IdentifiedNeed.ShelterReferral);
```
Comment (williamrosswhite) [inline ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs:111]: same here, though i'm out of context here

File: ess/src/API/EMBC.Tests.Integration.ESS/Resources/EvacuationTests.cs [hunk 4]
```diff
@@ -286,11 +282,7 @@ private EvacuationFile CreateTestFile(Evacuee primaryContact)
                     },
                     Type = NeedsAssessmentType.Preliminary,
                     Insurance = InsuranceOption.Yes,
-                    CanProvideClothing = true,
-                    CanProvideFood = true,
-                    CanProvideIncidentals = true,
-                    CanProvideLodging = true,
```
Comment (williamrosswhite) [inline ess/src/API/EMBC.Tests.Integration.ESS/Resources/EvacuationTests.cs:292]: same but not going to keep pointing it out

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs (hunk 2 above)
Comment (ytqsl) [reply ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs:74]: The Dynamics schema is not changing, so we'll still need to map to the existing fields

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs (hunk 3 above)
Comment (ytqsl) [reply ess/src/API/EMBC.ESS/Resources/Evacuations/Mappings.cs:111]: same

File: ess/src/API/EMBC.Tests.Integration.ESS/Resources/EvacuationTests.cs (hunk 4 above)
Comment (ytqsl) [reply ess/src/API/EMBC.Tests.Integration.ESS/Resources/EvacuationTests.cs:292]: same

File: ess/src/API/EMBC.ESS/Resources/Evacuations/Contract.cs (hunk 1 above)
Comment (ytqsl) [reply ess/src/API/EMBC.ESS/Resources/Evacuations/Contract.cs:171]: Actually, we can't make it a single field - it is 'grouped' in the front end, but eventually it is one or the other that will be enforced in the UI.

**Review by TimaQT — COMMENTED** [review]
lgtm

**Review by TimaQT — APPROVED** [review]

**Pull request merged**